__tests__/unit/backend-proxy.test.js

```diff
@@ -380,6 +380,42 @@ describe('Backend Proxy', () => {
 			expect(next).not.toHaveBeenCalled();
 		});
 
+		test('rewrites the location header if the domain matches the backend domain, but not the backend path', () => {
+			// Given
+			const middleware = backendProxy({
+				...baseOptions,
+				backend: baseOptions.backend + '/additional/path'
+			});
+			let relativeLocation = `/some-wonderful/location?here=there#my-id`;
+
+			const backendResponse = new EventEmitter();
+			backendResponse.headers = {
+				location: baseOptions.backend + relativeLocation
+			};
+
+			backendResponse.statusCode = 302;
+			backendResponse.pipe = jest.fn();
+
+			const response = {
+				header: jest.fn()
+			};
+
+			middleware(mockRequest, response, next);
+
+			// When
+			proxyRequest.emit('response', backendResponse);
```

Comment:
This test needs DRY-ing up a bit. theres some common setup being done in these location header tests that should be stuck in a beforeEach

Reply:
moved the repeated parts into a beforeEach and swapped `let` for `const` in a few places